dev/com.ibm.ws.ejbcontainer.bindings_fat/test-applications/ServerXMLBindingsWeb.war/src/com/ibm/ws/ejbcontainer/bindings/serverxml/bnd/web/ServerXMLBindingsTestServlet.java

```diff
@@ -0,0 +1,144 @@
+import com.ibm.ejb3x.HomeBindingName.ejb.LocalHomeBindingNameHome;
+import com.ibm.ejb3x.HomeBindingName.ejb.RemoteHomeBindingName;
+import com.ibm.ejb3x.HomeBindingName.ejb.RemoteHomeBindingNameHome;
+import com.ibm.ejb3x.SimpleBindingName.ejb.SimpleBindingName;
+import com.ibm.ejb3x.SimpleBindingName.ejb.SimpleBindingNameHome;
+
+import componenttest.app.FATServlet;
+
+/**
+ */
+@SuppressWarnings("serial")
+@WebServlet("/ServerXMLBindingsTestServlet")
+public class ServerXMLBindingsTestServlet extends FATServlet {
+
+    public void lookupServerXMLBindings() throws Exception {
+        InitialContext ctx = new InitialContext();
+
+        // Component-id
+        ComponentIDBndHome compIDbeanHome = (ComponentIDBndHome) ctx.lookup("ejblocal:ejb/ServerXMLMyEJB1#com.ibm.ejb3x.ComponentIDBnd.ejb.ComponentIDBndHome");
+        if (compIDbeanHome == null) {
+            fail("lookup component-id was null");
+        }
+        ComponentIDBnd compIDbean = compIDbeanHome.create();
+        if (compIDbean == null) {
+            fail("home.create() component-id was null");
+        }
+        System.out.println("Got bean, calling method");
+        if (compIDbean.foo() == null) {
+            fail("bean.method() for component-id was null");
+        }
+
+        // local-home-binding
+        LocalHomeBindingNameHome LHBbeanHome = (LocalHomeBindingNameHome) ctx.lookup("ejblocal:com/ibm/ejb3x/HomeBindingName/ejb/ServerXMLHomeBindingNameHome1");
+        if (LHBbeanHome == null) {
+            fail("lookup local-home-binding was null");
+        }
+        LocalHomeBindingName LHBbean = LHBbeanHome.create();
+        if (LHBbean == null) {
+            fail("home.create() for local-home-binding was null");
+        }
+        if (LHBbean.foo() == null) {
+            fail("bean.method() for local-home-binding was null");
+        }
+
+        // remote-home-binding
+        Object RHBlookup = ctx.lookup("com/ibm/ejb3x/HomeBindingName/ejb/ServerXMLHomeBindingNameHome3");
+        RemoteHomeBindingNameHome RHBbeanHome = (RemoteHomeBindingNameHome) PortableRemoteObject.narrow(RHBlookup, RemoteHomeBindingNameHome.class);
+        if (RHBbeanHome == null) {
+            fail("lookup for remote-home-binding was null");
+        }
+        RemoteHomeBindingName RHBbean = RHBbeanHome.create();
+        if (RHBbean == null) {
+            fail("lookup for remote-home-binding was null");
+        }
+        if (RHBbean.foo() == null) {
+            fail("lookup for remote-home-binding was null");
+        }
+
+        // simple-binding-name
+        SimpleBindingNameHome SBNbeanHome = (SimpleBindingNameHome) ctx.lookup("ejblocal:com/ibm/ejb3x/SimpleBindingName/ejb/ServerXMLSimpleBindingNameHome1");
+        if (SBNbeanHome == null) {
+            fail("lookup for simple-binding-name was null");
+        }
+        SimpleBindingName SBNbean = SBNbeanHome.create();
+        if (SBNbean == null) {
+            fail("lookup for simple-binding-name was null");
```

Comment:
There's some errant copy/paste issues with the assertion text here and with remote-home-binding below.  Also, is there any particular reason that you're not just using assertNotNull for all of these test points?

Reply:
I copy pasted those from another test where the message was very long so it was more readable with the if statements, but since I cut the message down `assertNotNull` does make more sense, changed it.